src/main/scala/com/fulcrumgenomics/vcf/VariantMask.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.vcf
+
+import com.fulcrumgenomics.FgBioDef._
+import htsjdk.samtools.SAMSequenceDictionary
+import htsjdk.variant.variantcontext.VariantContext
+import htsjdk.variant.vcf.VCFFileReader
+
+import scala.collection.mutable
+import scala.collection.JavaConversions.asScalaIterator
+
+object VariantMask {
+  /** Generate a variant mask from the provided VCF. */
+  def apply(path: PathToVcf): VariantMask = apply(new VCFFileReader(path.toFile))
+
+  /** Generate a variant mask from the provided VCF reader. */
+  def apply(reader: VCFFileReader): VariantMask = {
+    val dict = reader.getFileHeader.getSequenceDictionary
+    require(dict != null && dict.getSequences.size() > 0, "Generating a VariantMask requires VCFs contain contig lines.")
+    apply(reader.iterator(), dict)
+  }
+
+  /** Generates a VariantMask from the variants in the provided iterator. */
+  def apply(variants: Iterator[VariantContext], dict: SAMSequenceDictionary) = new VariantMask(variants, dict)
+}
+
+/**
+  * Simple mask that loads variants one reference sequence at a time and creates a compact
+  * representation allowing for rapid querying of whether or not positions are overlapped
+  * by one or more variants.
+  *
+  * @param variants a coordinate sorted iterator of variants
+  * @param dict the sequence dictionary for the reference
+  */
+class VariantMask(variants: Iterator[VariantContext], val dict: SAMSequenceDictionary) {
```

Comment:
It depends on usage.  If you're looking at a small set of intervals with truth, maybe not.  If you're looking at WGS data with dbSNP, several million entries?